presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -820,6 +823,19 @@ public void finishDelete(ConnectorSession session, ConnectorTableHandle tableHan
                 .map(RaptorColumnHandle.class::cast)
                 .map(ColumnInfo::fromHandle).collect(toList());
 
+        if (table.isTableSupportsDeltaDelete()) {
+            ImmutableMap.Builder<UUID, Pair<Optional<UUID>, Optional<ShardInfo>>> shardMapBuilder = new ImmutableMap.Builder<>();
+
+            fragments.stream()
+                    .map(fragment -> SHARD_DELTA_DELETE_CODEC.fromJson(fragment.getBytes()))
+                    .forEach(delta -> shardMapBuilder.put(delta.getOldShardUuid(), new Pair<Optional<UUID>, Optional<ShardInfo>>(delta.getOldDeltaDeleteShard(), delta.getNewDeltaDeleteShard())));
+            OptionalLong updateTime = OptionalLong.of(session.getStartTime());
+
+            log.info("Finishing delete for tableId %s (affected shardUuid: %s)", tableId, shardMapBuilder.build().size());
+            shardManager.replaceDeltaUuids(transactionId, tableId, columns, shardMapBuilder.build(), updateTime);
+            clearRollback();
+            return;
```

Comment:
- Use `else` for the following branch
- remove `return`
- move `clearRollback();` to the end